### PR TITLE
[Backport 2.x] Update imports for files refactored in core PR #8157 (#3003)

### PR DESCRIPTION
**Pull request by DarshitChanpura**
### Description
Backports https://github.com/opensearch-project/security/commit/37aacdc0ceab4462498c9bdf339e6e0b5d30ef80 from #3003 

Also adds integration tests to 2.x line.

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by DarshitChanpura** [comment]
Plugin install seems to be failing due to stale core artifact. Will wait for new artifact to be available before re-running these tasks.

The error below should not be thrown as that is the correct path for [Writeable.Reader](https://github.com/opensearch-project/OpenSearch/blob/83a4b6e9effc0e73ecba67e5dd9246e06b2c3df5/libs/core/src/main/java/org/opensearch/core/common/io/stream/Writeable.java#L166):
```
java.lang.NoClassDefFoundError: org/opensearch/core/common/io/stream/Writeable$Reader
```

**Comment by DarshitChanpura** [comment]
Backwards-compatibility also seems to be failing only for windows due to stale artifact issue too.

File: src/integrationTest/java/org/opensearch/security/http/JwtAuthenticationTests.java [hunk 1]
```diff
@@ -0,0 +1,270 @@
+/*
```
Comment (cwperks) [inline src/integrationTest/java/org/opensearch/security/http/JwtAuthenticationTests.java:1]: Should the files in `integrationTest` be added to 2.x? @peternied thoughts?

File: src/main/java/org/opensearch/security/ssl/SecureSSLSettings.java [hunk 2]
```diff
@@ -0,0 +1,130 @@
+/*
```
Comment (cwperks) [inline src/main/java/org/opensearch/security/ssl/SecureSSLSettings.java:1]: Looks like this class is being added, but not utilized. Can you remove this change and/or include an entire backport of https://github.com/opensearch-project/security/pull/2296?

File: scripts/integtest.sh [hunk 3]
```diff
@@ -101,7 +101,7 @@ then
 fi
 if [ -z "$COMMON_UTILS_VERSION" ]
 then
-  COMMON_UTILS_VERSION="2.9.0.0-SNAPSHOT"
+  COMMON_UTILS_VERSION="3.0.0.0-SNAPSHOT"
```
[the remark this answers is not in the document]
Comment (DarshitChanpura) [reply scripts/integtest.sh:104]: Yes, this was accidental.

File: src/main/java/org/opensearch/security/ssl/SecureSSLSettings.java (hunk 2 above)
Comment (DarshitChanpura) [reply src/main/java/org/opensearch/security/ssl/SecureSSLSettings.java:1]: I'll create a separate backport for that and remove it from here.

File: src/integrationTest/java/org/opensearch/security/http/JwtAuthenticationTests.java (hunk 1 above)
Comment (DarshitChanpura) [reply src/integrationTest/java/org/opensearch/security/http/JwtAuthenticationTests.java:1]: We should have integrationTests for 2.x line.

Comment (cwperks) [reply src/integrationTest/java/org/opensearch/security/http/JwtAuthenticationTests.java:1]: Are you able to run them? iirc there is more to the setup than backporting individual files in `src/integrationTest`?

**Comment by DarshitChanpura** [comment]
Backwards compatibility tests are broken because the CI task checks out 2.x and current HEAD at 2.x doesn;t contain the import changes. (This PR brings those changes)

**Comment by DarshitChanpura** [comment]
We either need to force-merge this to fix bwc tests or temporarily change the reference to current branch to `DarshitChanpura/backport/backport-3003-to-2.x/` and fix that once this PR is merged into 2.x

**Comment by RyanL1997** [comment]
I think this is the similar issue happened on main, and if this is the case I think [Craig's procedure](https://github.com/opensearch-project/security/pull/3003#issuecomment-1638904833) can be also applied to here. But yeah, instead of referencing to the current 2.x, reference to current fix's branch will definitely resolve the issue.

**Comment by cwperks** [comment]
bwc tests need to be revisited on the 2.x branch. I see a change in main that isn't in 2.x that could help the issue: https://github.com/opensearch-project/security/pull/2253

**Review by cwperks — APPROVED** [review]

**Comment by cwperks** [comment]
I think the bwc tests in 2.x are doing too much. It should be testing from the 1.3 -> branch of the PR

**Comment by cwperks** [comment]
@DarshitChanpura I believe you can update these lines: https://github.com/opensearch-project/security/blob/2.x/.github/workflows/ci.yml#L92-L93

The new values would be `2.9` and `current_branch` respectively

**Review by cwperks — APPROVED** [review]

**Review by RyanL1997 — APPROVED** [review]

**Pull request merged**